src/main/java/jp/classmethod/aws/gradle/lambda/AWSLambdaInvokeTask.java

```diff
@@ -105,12 +105,14 @@ private void setupPayload(InvokeRequest request) throws IOException {
 		if (payload instanceof File) {
 			File file = (File) payload;
 			str = Files.toString(file, Charsets.UTF_8);
+			request.setPayload(str);
 		} else if (payload instanceof Closure) {
 			Closure<?> closure = (Closure<?>) payload;
 			str = closure.call().toString();
-		} else {
+			request.setPayload(str);
+		} else if (payload instanceof String) {
```

Comment:
1. If `payload` is already a `String`, then you don't call `toString`.
2. Another problem is introduced here: if there exists a client that relies on a `toString` method to return JSON, then this client will fail now. If `payload` isn't a `File`, `Closure` or `String` then no payload is sent to lambda.